### PR TITLE
docs(swagger): add docs for enumName

### PR DESCRIPTION
**Pull request by nartc**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/docs.nestjs.com/blob/master/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] Other... Please describe:
```

## What is the new behavior?
Add documentations for `enumName`

Issue Number: https://github.com/nestjs/swagger/pull/412

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: content/recipes/swagger.md [hunk 1]
```diff
@@ -160,6 +160,117 @@ With `isArray` set to **true**, the `enum` can be selected as a **multi-select**
 
 <figure><img src="/assets/enum_query_array.gif" /></figure>
 
+#### Enums Schema
+
+By default, `enum` property will add a raw definition of [Enum](https://swagger.io/docs/specification/data-models/enums/) on the `parameter`.
```
Comment (johnbiundo) [inline content/recipes/swagger.md]: By default, `enum` property
->
By default, the `enum` property

File: content/recipes/swagger.md [hunk 2]
```diff
@@ -160,6 +160,117 @@ With `isArray` set to **true**, the `enum` can be selected as a **multi-select**
...
+    - breed:
+        type: 'string'
+        enum:
+          - Persian
+          - Tabby
+          - Siamese
+    ...
+```
+
+The above specification works fine for most cases. However, if you are utilizing a tool that takes the specification as **input** and generates **client-side** code, you might run into a problem with the generated code containing duplicated `enums`. Check the following snippet out:
```
Comment (johnbiundo) [inline content/recipes/swagger.md]: Check the following snippet out: 
->
Consider the following code snippet:

File: content/recipes/swagger.md [hunk 3]
```diff
@@ -160,6 +160,117 @@ With `isArray` set to **true**, the `enum` can be selected as a **multi-select**
...
+
+export class CatInformation {
+  ...
+   @ApiProperty({ enum: CatBreed, enumName: 'CatBreed' })
+   breed: CatBreed;
+   ...
+}
+```
+
+`enumName` allows `nestjs/swagger` to be able to turn `CatBreed` into its own `schema` which in turns makes `CatBreed` reusable. The specification will look like the following:
```
Comment (johnbiundo) [inline content/recipes/swagger.md]: `enumName` allows `nestjs/swagger` to be able to turn
->
`enumName` enables `nestjs/swagger` to turn

File: content/recipes/swagger.md [hunk 4]
```diff
@@ -160,6 +160,117 @@ With `isArray` set to **true**, the `enum` can be selected as a **multi-select**
...
+
+CatBreed:
+  type: string
+  enum:
+    - Persian
+    - Tabby
+    - Siamese
+```
+
+> info **Hint** Any **decorator** that takes in `enum` as a property will also take in `enumName`.
```
Comment (johnbiundo) [inline content/recipes/swagger.md]: takes in `enum` as a property will also take in `enumName`.
->
takes `enum` as a property will also take `enumName`.

**Comment by nartc** [comment]
> I made several small style/grammar suggestions.

Thanks. Changes have been made according to suggestions

**Pull request merged**